apps/site/lib/site_web/views/event_view.ex

```diff
@@ -59,42 +61,70 @@ defmodule SiteWeb.EventView do
   @doc "Nicely renders the duration of an event, given two DateTimes."
   @spec render_event_duration(
           NaiveDateTime.t() | DateTime.t(),
-          NaiveDateTime.t() | DateTime.t() | nil
-        ) ::
+          NaiveDateTime.t() | DateTime.t() | nil,
           String.t()
-  def render_event_duration(start_time, end_time)
+        ) :: String.t() | %{date: String.t(), time: String.t()}
+  def render_event_duration(start_time, end_time, style)
 
-  def render_event_duration(start_time, nil) do
+  def render_event_duration(start_time, nil, style) do
     start_time
     |> maybe_shift_timezone
-    |> do_render_event_duration(nil)
+    |> do_render_event_duration(nil, style)
   end
 
-  def render_event_duration(start_time, end_time) do
+  def render_event_duration(start_time, end_time, style) do
     start_time
     |> maybe_shift_timezone
-    |> do_render_event_duration(maybe_shift_timezone(end_time))
+    |> do_render_event_duration(maybe_shift_timezone(end_time), style)
   end
 
-  defp do_render_event_duration(start_time, nil) do
-    "#{pretty_date(start_time, "{WDshort}, {Mshort} {D}, {YYYY}")} \u2022 #{
-      format_time(start_time)
-    }"
+  @spec do_render_event_duration(
+          NaiveDateTime.t() | DateTime.t(),
+          NaiveDateTime.t() | DateTime.t() | nil,
+          String.t()
+        ) :: String.t() | %{date: String.t(), time: String.t()}
+  defp do_render_event_duration(start_time, nil, style) do
+    if style === "list" do
```

Comment:
Probably should be an atom `:list` since you're using it as a way of distinguishing these two cases. However, maybe it's worth making two separate functions instead? `render_event_duration_calendar` and `render_event_duration_list`?
```elixir
def render_event_duration_list(start_time, end_time) do
  calendar = render_event_duration_calendar(start_time, end_time)
  "#{calendar.date} \u2022 #{calendar.time}"
end
...
```

Reply:
I'm trying to wrap my head around whether this would do it... dates are formatted differently in the calendar and list views

![Screen Shot 2021-04-15 at 11 32 48 AM](https://user-images.githubusercontent.com/69368883/114896416-50460a00-9dde-11eb-99ec-e790522261ec.png)
![Screen Shot 2021-04-15 at 11 32 29 AM](https://user-images.githubusercontent.com/69368883/114896455-5936db80-9dde-11eb-804a-04ce5354eed6.png)

Reply:
I mean the actual text itself is truncated differently, but I can pass the timex formatter as a string.  So maybe that'll work